modules/mbedtls/CMakeLists.txt

```diff
@@ -11,6 +11,18 @@ zephyr_interface_library_named(mbedTLS)
       message(WARNING "No entropy device on the system, using fake entropy source!")
     endif()
 
+    if(CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG)
+      message(WARNING "
+        Warning: non cryptographycally secure sources are enabled for psa_generate_random().
+        This is meant to be used only for tests, not in production!")
+    else()
+      if(CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG AND NOT CONFIG_CSPRNG_ENABLED)
+        message(WARNING "
+          MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG is set but there is
+          no CSPRNG enabled.")
+      endif()
```

Comment:
Couldn't we just make the build fail in this case? If `EXTERNAL_RNG` is enabled, CSPRNG is not enabled, nor is `RNG_ALLOW_NON_CSPRNG`, then `psa_crypto_init()` will always fail right?